### PR TITLE
Harden cloud sync: simplify S3 config, safe backend switching, backup + provider cleanup

### PR DESCRIPTION
**Pull request by ericgriffin**
## Summary

This PR began as an S3 config cleanup and grew to cover several related cloud-sync
improvements. It delivers four distinct areas:

### 1. Simplified S3 configuration with auto-detected region
- Shrinks the S3 config form to four visible fields (Endpoint, Bucket, Access Key ID, Secret Access Key); Region, Key Prefix, and Path-style move into a collapsed Advanced section.
- Region is resolved automatically: manual entry wins, otherwise derived from the endpoint hostname (R2 to auto, B2/DO/Wasabi/Scaleway/AWS parsed from host, unknown to `us-east-1`).
- `S3ApiClient` self-heals wrong regions: a rejected request carrying an `x-amz-bucket-region` header or `AuthorizationHeaderMalformed` hint is re-signed and replayed once (preserving any dualstack host variant); the correction is persisted by the provider and adopted by the form during Test Connection.
- `AuthorizationHeaderMalformed` errors now explain the region problem instead of reporting "Access denied".
- Endpoint URL is now required, with assorted form-clarity fixes and provider-neutral wording.

### 2. Hardened backend switching (schema v81)
Switching sync backends (e.g. S3 to iCloud) previously carried the global sync cursor
across, silently bypassing the first-contact merge guard and leaving no signal for
devices still on the old backend.
- Scope the sync cursor per provider (schema v81, `last_sync_provider`) so a switched-to backend triggers first-contact confirmation and a full reconcile instead of replaying a foreign baseline.
- Confirm before switching away from a backend that has sync history.
- Write a "library moved" marker to the old backend so straggler devices learn where the library went (advisory, fail-open).
- Offer to delete orphaned old-backend data after the first successful sync on the new backend.
- Cross-backend epoch isolation (self-heal + adoption-halt).

### 3. Cloud backup off by default and coupled to sync state
- `cloudBackupEnabled` now defaults to `false` (was `true`) and is tied to sync being enabled.
- Backup location moved below frequency and retention in the settings UI.
- Conflicting-key writes are ordered so a crash mid-update can only leave a valid (both-off) state, never cloud-backup-on with a custom location.

### 4. Google Drive provider hidden until implemented
- Google Drive is removed from the provider picker until its integration is complete; the `CloudProviderType` and provider plumbing remain in place.
- A persisted `googledrive` selection is treated as no provider so the page can never show a half-wired backend.

New user-facing strings are localized into all 10 non-English locales. The stored
`S3Config` shape is unchanged (no migration); the sync schema bumps to v81 for
`last_sync_provider`.

Spec: docs/superpowers/specs/2026-06-12-s3-config-simplification-design.md
Plan: docs/superpowers/plans/2026-06-12-s3-config-simplification.md

## Test Plan
- [x] `deriveRegion` unit tests cover every provider hostname pattern plus fallback/edge cases.
- [x] S3 API client tests cover hint adoption (header and XML body), AWS host rebuild + dualstack preservation, single-replay guard, callback semantics, corrected-region stickiness.
- [x] Provider tests cover correction persistence without client invalidation and `testConnection` pass-through without persistence.
- [x] S3 widget tests cover the collapsed Advanced section, live auto-detect helper, manual override, derived-region save, detection snackbar and field adoption, existing-config load.
- [x] Backend-switch coverage: per-provider cursor scoping, v81 migration, moved-marker read/write + old-backend cleanup paths, switch confirmation dialog, epoch self-heal/adoption-halt, first-contact guard.
- [x] Backup coverage: default-off, sync coupling, settings reordering, atomic key swap.
- [x] Full suite green; `flutter analyze` and `dart format` clean.
- [x] Manual verification against a real provider (MinIO/B2/R2/AWS) and a two-device backend switch recommended before release.

Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: lib/core/services/cloud_storage/s3/s3_api_client.dart [hunk 1]
```diff
@@ -194,6 +205,16 @@ class S3ApiClient {
...
       );
+      if (response.statusCode >= 300) {
+        final corrected = await _replayWithRegionHint(
+          response,
+          method,
+          key,
+          queryParams,
+          body,
+        );
+        if (corrected != null) return corrected;
```
[the remark this answers is not in the document]
Comment (ericgriffin) [reply lib/core/services/cloud_storage/s3/s3_api_client.dart]: Fixed in d48648531d3: the replay response now falls through to the standard <500 check instead of returning early, so a 5xx after a region correction gets the normal delayed retry — which re-signs with the already-adopted corrected region. Covering test added (301 hint, then 500 replay, then 200 retry: completes in 3 requests; onRegionCorrected stays unfired since the replay itself did not succeed).

File: lib/features/settings/presentation/pages/s3_config_page.dart [hunk 2]
```diff
@@ -121,18 +142,33 @@ class _S3ConfigPageState extends ConsumerState<S3ConfigPage> {
...
     try {
       await ref
           .read(s3StorageProviderInstanceProvider)
-          .testConnection(_buildConfig());
-      _showSnack(successMessage);
+          .testConnection(
+            _buildConfig(),
+            onRegionCorrected: (region) => detectedRegion = region,
+          );
+      final detected = detectedRegion;
```
[the remark this answers is not in the document]
Comment (ericgriffin) [reply lib/features/settings/presentation/pages/s3_config_page.dart:162]: Fixed in bdf86d65dc4: a mounted check now sits immediately after the await, before the controller write. One nuance found while reproducing: the method's catch-all was silently swallowing the disposed-controller AssertionError, so this never actually crashed — but relying on that was worse than the bug. A regression test pins disposal-during-probe staying clean.

File: lib/features/settings/presentation/pages/s3_config_page.dart [hunk 3]
```diff
@@ -121,18 +143,33 @@ class _S3ConfigPageState extends ConsumerState<S3ConfigPage> {
...
+      if (detected != null) {
+        _regionController.text = detected;
+        _showSnack(l10n.settings_s3Config_test_regionDetected(detected));
+      } else {
+        _showSnack(l10n.settings_s3Config_test_success);
+      }
```
[the remark this answers is not in the document]
Comment (ericgriffin) [reply lib/features/settings/presentation/pages/s3_config_page.dart:168]: Same issue as the thread on line 159 — fixed in bdf86d65dc4 with a mounted guard after the await, plus a disposal-mid-probe regression test.

File: lib/core/services/cloud_storage/s3/s3_api_client.dart [hunk 4]
```diff
@@ -164,8 +175,8 @@ class S3ApiClient {
     if (_config.isAws) {
       scheme = 'https';
       host = _config.pathStyle
-          ? 's3.${_config.region}.amazonaws.com'
-          : '${_config.bucket}.s3.${_config.region}.amazonaws.com';
+          ? 's3.$_region.amazonaws.com'
+          : '${_config.bucket}.s3.$_region.amazonaws.com';
```
[the remark this answers is not in the document]
Comment (ericgriffin) [reply lib/core/services/cloud_storage/s3/s3_api_client.dart:179]: Fixed in bdf86d65dc4: _target() now routes regional AWS endpoint hosts (s3.{r}, s3.dualstack.{r}, legacy s3-{r}) through the effective region, so a server hint moves the replay to the corrected regional host. The global s3.amazonaws.com is deliberately left untouched — it routes cross-region via DNS and only needs the SigV4 scope fix. Tests cover the wrong-regional-endpoint correction and that non-AWS hosts are never rewritten.

File: lib/features/settings/presentation/pages/cloud_sync_page.dart [hunk 5]
```diff
@@ -919,13 +919,18 @@ class CloudSyncPage extends ConsumerWidget {
...
+        : '';
     final confirmed = await showDialog<bool>(
       context: context,
       builder: (context) => AlertDialog(
         title: const Text('Sign Out?'),
-        content: const Text(
+        content: Text(
           'This will disconnect from the cloud provider. '
-          'Your local data will remain intact.',
+          'Your local data will remain intact.$backupWarning',
```
[the remark this answers is not in the document]
Comment (ericgriffin) [reply lib/features/settings/presentation/pages/cloud_sync_page.dart]: Fixed in bdf86d65dc4 — good catch that settings_cloudSync_signOutDialog_* keys already existed with translations but were never wired. The dialog title, content, and both buttons now use them, so the appended backup warning stays in the same locale as the rest of the dialog.

File: docs/superpowers/specs/2026-06-12-s3-config-simplification-design.md [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+existing validators, the busy state, snackbars, and save/remove flows are
+unchanged.
+
+Amended 2026-06-12 after manual review:
+
+- The Endpoint URL is REQUIRED — no blank-means-AWS assumption in the
+  form. AWS users enter `https://s3.amazonaws.com` (or a regional
+  endpoint); the guidance is an in-field hint ("For Amazon S3, enter
+  `https://s3.amazonaws.com`") rather than helper text below the field,
+  which read as ambiguous between stacked fields.
```
[the remark this answers is not in the document]
Comment (ericgriffin) [reply docs/superpowers/specs/2026-06-12-s3-config-simplification-design.md]: Fixed in bdf86d65dc4: the spec amendment now records the later provider-neutrality decision (generic example hint) instead of the superseded AWS-specific copy, so spec and implementation match.

File: lib/features/settings/presentation/pages/cloud_sync_page.dart [hunk 7]
```diff
@@ -414,16 +414,9 @@ class CloudSyncPage extends ConsumerWidget {
...
-          title: 'Google Drive',
-          subtitle: 'Sync via Google Drive',
-          icon: Icons.cloud_circle,
-          isSelected: selectedProvider == CloudProviderType.googledrive,
-          isAvailable: true,
-        ),
+        // Google Drive is hidden until its integration is fully
+        // implemented; the CloudProviderType and provider plumbing remain
+        // so re-enabling is just restoring this tile.
         _buildS3ProviderTile(context, ref, selectedProvider),
```
[the remark this answers is not in the document]
Comment (ericgriffin) [reply lib/features/settings/presentation/pages/cloud_sync_page.dart:429]: Fixed in 29be6bf3396 — the page now coerces `CloudProviderType.googledrive` to null before deriving `hasProvider`/the tile-selection state, so a persisted GD choice or the SyncRepository `orElse: googledrive` fallback no longer leaves Sync Now enabled with no visible tile. Regression test asserts both the disabled Sync Now and the visible "select a provider" hint when googledrive is the persisted selection.

File: lib/core/services/cloud_storage/s3_storage_provider.dart [hunk 8]
```diff
@@ -229,7 +245,30 @@ class S3StorageProvider
...
+    final config = _cachedConfig;
+    if (config == null || config.region == region) return;
+    final updated = config.copyWith(region: region);
+    _cachedConfig = updated;
+    try {
+      await _store.save(updated);
+      _log.info('Persisted server-corrected S3 region: $region');
+    } catch (e) {
+      _log.warning('Could not persist corrected S3 region: $e');
     }
```
[the remark this answers is not in the document]
Comment (ericgriffin) [reply lib/core/services/cloud_storage/s3_storage_provider.dart:296]: Fixed in ee4f5b9918a — `_persistInFlight` tracks the in-flight region save; `signOut` and `saveConfig` await it before touching the store, so the `clear()`-after-`save()` reordering that would resurrect credentials post-sign-out cannot occur. A generation guard inside `_persistCorrectedRegion` also drops the post-save cache write if `_invalidate()` lands mid-call (defensive belt + braces for parallel saveConfig paths). Regression parks the save at a gate, starts `signOut`, asserts ordering (`save-begin` → `save-end` → `clear`), and confirms both the store and `_cachedConfig` end empty.

File: lib/features/backup/data/repositories/backup_preferences.dart [hunk 9]
```diff
@@ -36,7 +36,7 @@ class BackupPreferences {
       lastBackupTime: lastBackupMs != null
           ? DateTime.fromMillisecondsSinceEpoch(lastBackupMs)
           : null,
-      cloudBackupEnabled: _prefs.getBool(_cloudBackupEnabledKey) ?? true,
+      cloudBackupEnabled: _prefs.getBool(_cloudBackupEnabledKey) ?? false,
       backupLocation: _prefs.getString(_backupLocationKey),
     );
```
[the remark this answers is not in the document]
Comment (ericgriffin) [reply lib/features/backup/data/repositories/backup_preferences.dart:41]: Intentional — that flip is the headline of ab1dfc99a61 ("cloud backup off by default and coupled to sync state"). The new model is opt-in cloud backup gated on having a sync provider, which is also why `_confirmSignOut` and S3 config removal now call `disableCloudBackup()` automatically. A silent migration that re-enabled cloud backup for existing installs would push them back into the implicit-on state the commit was unwinding; existing installs that were uploading via the implicit-on default still keep their local backups, the cloud upload simply stops until the user flips the switch in Backup Settings. Happy to add a one-shot migration if you'd rather preserve the prior implicit default for upgraders, but my read is the explicit-opt-in is the stronger long-term shape.

File: lib/features/backup/presentation/providers/backup_providers.dart [hunk 10]
```diff
@@ -57,13 +57,28 @@ class BackupSettingsNotifier extends StateNotifier<BackupSettings> {
...
     await _prefs.setCloudBackupEnabled(value);
-    state = state.copyWith(cloudBackupEnabled: value);
+    if (value) await _prefs.setBackupLocation(null);
+    state = _prefs.getSettings();
   }
 
   Future<void> setBackupLocation(String? path) async {
     await _prefs.setBackupLocation(path);
+    if (path != null) await _prefs.setCloudBackupEnabled(false);
+    state = _prefs.getSettings();
```
[the remark this answers is not in the document]
Comment (ericgriffin) [reply lib/features/backup/presentation/providers/backup_providers.dart:77]: Fixed in 5d5e8c052ae: `setCloudBackupEnabled`/`setBackupLocation` now clear the conflicting key BEFORE setting the new one, so a crash between the two awaited writes can only persist a valid (both-off) state, never the invalid cloud-backup-on + custom-location combination. A recording-subclass regression test snapshots the persisted combination after every underlying write and asserts the enabled+custom-location pair never appears, even transiently.

File: lib/core/services/cloud_storage/s3/s3_api_client.dart [hunk 11]
```diff
@@ -227,6 +251,63 @@ class S3ApiClient {
...
+    final match = _awsRegionalHostPattern.firstMatch(
+      endpointHost.toLowerCase(),
+    );
+    if (match == null || match.group(1) == _region) return endpointHost;
+    return 's3.$_region.amazonaws.com';
+  }
+
+  static final _awsRegionalHostPattern = RegExp(
+    r'^s3[.-](?:dualstack\.)?([a-z0-9-]+)\.amazonaws\.com$',
+  );
```
[the remark this answers is not in the document]
Comment (ericgriffin) [reply lib/core/services/cloud_storage/s3/s3_api_client.dart:298]: Fixed in 5d5e8c052ae: `_effectiveCustomHost` now captures the `dualstack.` segment in `_awsRegionalHostPattern` and re-emits it when rebuilding the corrected host, so `s3.dualstack.<region>.amazonaws.com` stays dualstack after a correction (was collapsing to the plain `s3.<region>` host). Regression test: a dualstack endpoint hit with a 301 region hint replays against `s3.dualstack.<corrected-region>.amazonaws.com`. The legacy dash form (`s3-<region>`, which has no dualstack variant) still normalizes to the modern dot host.
